app/frontend/src/features/profile/ProfileForm.tsx

```diff
@@ -87,12 +93,34 @@ export default function EditProfileForm() {
       {user ? (
         <>
           <form onSubmit={onSubmit}>
+
+
+            <Controller
+                name={'avatarUrl'}
+                control={control}
+                render={({value}) => {
+                  return <AvatarInput
+                      id={'profile-picture'}
+                      name={'image'}
+                      src={value}
+                      onChange={async (file) => {
+                          // TODO move media endpoint to config
+                        const response = await service.api.uploadFile(file);
+                        const thumbnailUrl = 'https://dev-user-media.coucher.org/media/img/avatar/' + response.key + '.jpg';
+                        setValue('avatarUrl',thumbnailUrl)
+                      }}
+                  />
```

Comment:
I think you could move this all into the AvatarInput component. Not sure if others agree or not.

That way you don't have to specify onChange and there would be one place to change the upload logic.

Reply:
Additionally, if you use a [react-query mutation](https://react-query.tanstack.com/guides/mutations) you'll get free loading state

Reply:
+1 otherwise every time we have to upload, this code needs to be copied again (basically what Lucas said)

Reply:
I totally agree, I'll move this logic into the ImageInput component.
I'm not sure how to avoid having the onChange since I need to use the setValue function to set the thumbnail url on the form, right?